docs/reference/troubleshooting/troubleshooting-searches.asciidoc

```diff
@@ -0,0 +1,240 @@
+data. This can happen when there is a data ingestion issue. For example, the
+data may have been indexed to a data stream or index with another name.
+
+Use the <<search-count,count API>> to retrieve the number of documents in a data
+stream or index. Check that `count` in the response is not 0.
+
+////
+[source,console]
+----
+PUT my-index-000001
+----
+////
+
+[source,console]
+----
+GET /my-index-000001/_count
+----
+//TEST[continued]
+
+NOTE: When getting no search results in {kib}, check that you have selected the
+correct data view and a valid time range. Also, ensure the data view has been
+configured with the correct time field.
+
+[discrete]
+[[troubleshooting-searches-field-exists]]
+=== Ensure the field exists and has data
+
+Querying a field that does not exist or has no data will not return any results.
+Use the <<query-dsl-exists-query,`exists` query>> to check whether there are
+documents that contain an indexed value for a field. Check that `count` in the
+response is not 0.
+
+[source,console]
+----
+GET /my-index-000001/_count
+{
+  "query": {
+    "exists": {
+      "field": "my-field"
+    }
+  }
+}
+----
+//TEST[continued]
+
+If the field exists, and is not of type `text`, use a
+<<search-aggregations-bucket-terms-aggregation,terms aggregation>> to retrieve
+the field's most common values:
+
+[source,console]
+----
+GET /my-index-000001/_search?filter_path=aggregations
+{
+  "size": 0,
+  "aggs": {
+    "top_values": {
+      "terms": {
+        "field": "my-field",
+        "size": 10
+      }
+    }
+  }
+}
+----
+//TEST[continued]
+
+If the field does not exist or have any values, check the data ingestion
+process. The field may have a different name.
+
+[discrete]
+[[troubleshooting-searches-fieldcaps-mappings]]
+=== Check field capabilities and mappings
+
+When the field you're querying exists, but you get no or unexpected search
+results, check the capabilities and <<mapping,mapping>> of the field. 
+
+Use the <<search-field-caps,field capabilities API>> to retrieve the
+capabilities of a field. This will return a field's type and tell you if it is
+searchable and aggregatable:
+
+[source,console]
+----
+GET /my-index-000001/_field_caps?fields=my-field
+----
+//TEST[continued]
+
+A field's capabilities are determined by its mapping. To retrieve the mapping,
+use the <<indices-get-mapping,get mapping API>>:
+
+[source,console]
+----
+GET /my-index-000001/_mappings
+----
+//TEST[continued]
+
+For `text` fields, pay attention to the analyzer that may have been configured.
+You can use the <<indices-analyze,analyze API>> to check how a field's analyzer
+processes values and query terms:
+
+[source,console]
+----
+GET /my-index-000001/_analyze
+{
+  "field" : "my-field",
+  "text" : "this is a test"
+}
+----
+//TEST[continued]
+
+<<updating-field-mappings,Changing the mapping of a field>> generally requires
+creating a new index with the correct mappings. Next, reindex the data into that
```

Comment:
not for runtime fields :) they can be removed at all times. But there isn't yet support for a text runtime field, so what you wrote suits well this specific scenario.

Reply:
I'll simplify this paragraph to simply point to the "Change the mapping of an existing field" documentation